apepay/__init__.py

```diff
@@ -58,9 +58,44 @@ def normalize_address(cls, value: Any) -> AddressType:
 
     @property
     def contract(self) -> ContractInstance:
-        # TODO: Fix this
-        return self.project_manager.StreamManager.at(self.address)
-        # return self.chain_manager.contracts.instance_at(self.address)
+        try:
+            return self.chain_manager.contracts.instance_at(self.address)
+        except Exception:
+            return self.project_manager.StreamManager.at(self.address)
+
+    def _get_contract_receipt(self, start_block=0, stop_block=None) -> ReceiptAPI:
+        if stop_block is None:
+            stop_block = self.chain_manager.blocks.head.number
+
+        mid_block = (stop_block - start_block) // 2 + start_block
+        # NOTE: biased towards mid_block == start_block
+
+        if start_block == mid_block:
+            for tx in self.chain_manager.blocks[mid_block].transactions:
+                if tx.receipt.contract_address == self.address:
+                    return tx.receipt
+
+            raise  # cannot find receipt
+
+        # HACK: until https://github.com/ApeWorX/ape/issues/1540
+        # elif self.provider.get_code(self.address, block_id=mid_block)
+        elif self.provider.web3.eth.get_code(self.address, block_identifier=mid_block):
+            return self._get_contract_receipt(
+                start_block=start_block, stop_block=mid_block
+            )
+
+        else:
+            return self._get_contract_receipt(
+                start_block=mid_block + 1, stop_block=stop_block
+            )
+
+    @cached_property
+    def creation_block(self) -> int:
```

Comment:
Hack for https://github.com/ApeWorX/ape/issues/1062 to speed up this:
https://github.com/ApeWorX/ApePay/pull/17/files#diff-0eed34615ea15b0b1032af02a3019ad4398e3850fa1a5797402a70da7f48edb6R275-R277

Reply:
Can remove all of this with https://github.com/ApeWorX/ape/pull/1548

Reply:
- [ ] Release ape 0.6.14 and bump here